child_compassion/model/child_compassion.py

```diff
@@ -504,13 +547,115 @@ def child_remove_from_typo3(self, cr, uid, ids, context=None):
             with sftp.cd(path):
                 sftp.put(filename)
 
-        self._typo3_scripts_fetch(scripts_url, api_key, "upd_db")
+        return self._typo3_scripts_fetch(scripts_url, api_key, "upd_db")
+
+    def child_add_to_typo3(self, cr, uid, ids, context=None):
+        reload(sys)
+        sys.setdefaultencoding('UTF8')
+
+        for child in self.browse(cr, uid, ids, context):
+
+            child_gender = self._get_gender(cr, uid, child.gender, context)
+            child_image = child.code+"_f.jpg,"+child.code+"_h.jpg"
+            child_birth_date = calendar.timegm(
+                datetime.strptime(child.birthdate, DF).utctimetuple())
+
+            child_desc_de = child.desc_de.replace('\'', '\'\'')
+            child_desc_fr = child.desc_fr.replace('\'', '\'\'')
+
+            # German description (parent)
+            self._update_request_to_typo3(
+                cr, uid,
+                "insert into "
+                "tx_drechildpoolmanagement_domain_model_children"
+                "(child_key, child_name_full, child_name_personal,"
+                "child_gender, child_biography,"
+                "l10n_parent,image,child_birth_date) "
+                "values ('{}','{}','{}','{}','{}','{}','{}','{}');".format(
+                    child.code, child.name, child.firstname, child_gender,
+                    child_desc_de, 0, child_image, child_birth_date),
+                context)
+
+            parent_id = self._get_typo3_child_id(cr, uid, child.code, context)
+
+            # French description
+            self._update_request_to_typo3(
+                cr, uid,
+                "insert into "
+                "tx_drechildpoolmanagement_domain_model_children"
+                "(child_key,child_name_full,child_name_personal,"
+                "child_gender,child_biography,"
+                "l10n_parent,image,child_birth_date) "
+                "values ('{}','{}','{}','{}','{}','{}','{}','{}');".format(
+                    child.code, child.name, child.firstname,
+                    child_gender, child_desc_fr, parent_id,
+                    child_image, child_birth_date),
+                context)
+
+        self._add_child_pictures_to_typo3(cr, uid, ids, context)
+
+    def _add_child_pictures_to_typo3(self, cr, uid, ids, context=None):
+        for child in self.browse(cr, uid, ids, context):
+            head_image = child.code+"_h.jpg"
+            full_image = child.code+"_f.jpg"
+
+            if not(child.portrait and child.fullshot):

```

Comment:
I would move this check in method `put_child_on_internet` of the wizard, so that if a child is missing its picture, it is not inserted in the typo3 database.